test/repositories/EmailRepositorySpec.scala

```diff
@@ -34,29 +36,31 @@ class EmailRepositorySpec extends SpecBase {
     repository.collection.drop().toFuture().map(_ => ())
   }
 
-  "return 'true' if an update has been performed on a record" in {
-    val eori = "SomeEori"
-    val notificationEmail = NotificationEmail("some@email.com", DateTime.now(), None)
+  "successfully retrive email from repository" in new Setup {
 
-    val undeliverableInformationEvent: UndeliverableInformationEvent = UndeliverableInformationEvent(
-      "some-id",
-      "some event",
-      "some@email.com",
-      "detected",
-      Some(12),
-      Some("unknown reason"),
-      s"HMRC-CUS-ORG~EORINumber~$eori",
-      Some("sdds")
-    )
+    val notificationEmail = NotificationEmail("123", DateTime.now, Some(undeliverableInformation))
 
-    val undeliverableInformation: UndeliverableInformation =
-      UndeliverableInformation(
-        "some-subject",
-        "some-event-id",
-        "some-group-id",
-        DateTime.now(),
-        undeliverableInformationEvent
-      )
+    await(for {
+      result <- repository.set(eori, notificationEmail)
+    } yield {
+      result mustBe SuccessfulEmail
+    })
+  }
+
+  "fail to retrieve email from repository after setting bad email" in new Setup {
+    val mockEmailRepository = mock[DefaultEmailRepository]
+    when(mockEmailRepository.set(any(), any())).thenReturn(Future.successful(FailedToRetrieveEmail))
+    val notificationEmail = NotificationEmail("123", DateTime.now, Some(undeliverableInformation))
+
+    await(for {
```

Comment:
Please remove await. Better to use map as below

    mockEmailRepository.set(eori, notificationEmail).map {
      result => result mustBe FailedToRetrieveEmail
    }